File: solutions/217 - Contains Duplicate/one_pass_hash_set.java
import java.util.*;
class Solution {
    public boolean containsDuplicate(int[] nums) {
        // Time Complexity: O(N) to build hash set and loop through the array.
        // Space Complexity: O(N) for the hash set.
        Set<Integer> set = new HashSet<>();
        for (int i : nums) {
            if (set.contains(i)) {
                return true;
            }
            set.add(i);
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter array elements:");
        String line = sc.nextLine();

        String[] parts = line.trim().split("\\s+");
        int[] nums = new int[parts.length];
        for (int i = 0; i < parts.length; i++) {
            nums[i] = Integer.parseInt(parts[i]);
        }

        Solution solution = new Solution();
        boolean result = solution.containsDuplicate(nums);

        System.out.println("Contains duplicate? " + (result ? "True" : "False"));

        sc.close();
    }
}

File: solutions/217 - Contains Duplicate/one_pass_hash_set.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <sstream>
using namespace std;
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        // Time Complexity: O(N) to build the hash set.
        // Space Complexity: O(N) for the hash set.
        unordered_set<int> hashSet;
        for (int i : nums)
        {
            hashSet.insert(i);
        }
        return hashSet.size() != nums.size();
    }
};

int main() {
    cout << "Enter array elements: ";
    string line;
    getline(cin, line);

    stringstream ss(line);
    vector<int> nums;
    int x;
    while (ss >> x) {
        nums.push_back(x);
    }

    Solution solution;
    bool result = solution.containsDuplicate(nums);

    cout << "Contains duplicate? " << (result ? "True" : "False") << endl;

    return 0;
}